tests/end_to_end/end_to_end.cpp

```diff
@@ -248,113 +249,185 @@ TEST_CASE("droppacket-jit", "[droppacket_jit]")
     hook.detach();
 }
 
-TEST_CASE("droppacket-interpret", "[droppacket_interpret]")
+void
+divide_by_zero_test(ebpf_execution_type_t execution_type)
 {
     _test_helper test_helper;
 
     ebpf_handle_t program_handle;
-    const char* error_message = NULL;
     ebpf_handle_t map_handle;
     uint32_t count_of_map_handle = 1;
     uint32_t result = 0;
-
-    program_information_provider_t xdp_program_information(EBPF_PROGRAM_TYPE_XDP);
+    const char* error_message = NULL;
 
     single_instance_hook_t hook;
+    program_information_provider_t xdp_program_information(EBPF_PROGRAM_TYPE_XDP);
 
     REQUIRE(
         (result = ebpf_api_load_program(
-             SAMPLE_PATH "droppacket.o",
+             SAMPLE_PATH "divide_by_zero.o",
              "xdp",
-             EBPF_EXECUTION_INTERPRET,
+             execution_type,
              &program_handle,
              &count_of_map_handle,
              &map_handle,
              &error_message),
          error_message ? printf("ebpf_api_load_program failed with %s\n", error_message) : 0,
          ebpf_api_free_string(error_message),
-         error_message = NULL,
+         error_message = nullptr,
          result == EBPF_SUCCESS));
 
     REQUIRE(hook.attach(program_handle) == EBPF_SUCCESS);
 
     auto packet = prepare_udp_packet(0);
 
-    uint32_t key = 0;
-    uint64_t value = 1000;
-    REQUIRE(
-        ebpf_api_map_update_element((ebpf_handle_t)1, sizeof(key), (uint8_t*)&key, sizeof(value), (uint8_t*)&value) ==
-        EBPF_SUCCESS);
-
     // Test that we drop the packet and increment the map
     xdp_md_t ctx{packet.data(), packet.data() + packet.size()};
+
     REQUIRE(hook.fire(&ctx, &result) == EBPF_SUCCESS);
-    REQUIRE(result == 2);
+    REQUIRE(result == -1);
```

Comment:
Can you at least add a comment explaining why "-1" is the expected result?  Is that the result of a computation, or an error code, or what?

Reply:
uBPF returns -1 when the program hits divide by zero. It's hard-coded into the divide-by-zero handler.

Reply:
Sure, but I can't tell that by reading the source code.   Instead of using a magic number, either use a #define, enum, or at minimum a // comment in the code.

Reply:
Commit 2dd2287ca5e3ff0721d6f0e48451de6b26657ade adds the following comment:
```
// uBPF returns -1 when the program hits a divide by zero error.
```

Reply:
I take it uBPF does not define any constant?   Could we get ubpf to add a constant so depdendent projects like this one don't need to hard code magic numbers?  If not, it's ok with that comment.